v7.3 Changelog (#5871)
mattermost/docs: source/install/self-managed-changelog.md

See the [changelog in progress](https://bit.ly/2nK3cVf) for the upcoming release. See the [Legacy Self-Hosted Mattermost Changelog](legacy-self-hosted-changelog) for details on all Mattermost self-hosted releases prior to v6.0. 

Latest Mattermost Releases:
- [Release v7.3 - Feature Release](#release-v7-2-feature-release)
- [Release v7.2 - Feature Release](#release-v7-2-feature-release)
- [Release v7.1 - Extended Support Release](#release-v7-1-extended-support-release)
- [Release v7.0 - Major Release](#release-v7-0-major-release)
- [Release v6.3 - Extended Support Release](#release-v6-3-extended-support-release)

## Release v7.3 - [Feature Release](https://docs.mattermost.com/upgrade/release-definitions.html#feature-release)

**v7.3.0 release day: 2022-09-16**

### Important Upgrade Notes

>  - Boards is moving from a channel-based to a role-based permissions system. The migration will happen automatically, but your administrator should perform a backup prior to the upgrade. We removed workspaces, so if you were a member of many boards prior to migration, they will now all appear under the same sidebar. Please see [this document](https://docs.mattermost.com/welcome/whats-new-in-v72.html) for more details.

Reviewer: Looks good to me! @sbishel Anything else you want to add here?

 - Fixed an issue where the Threads view displayed as unread even if there were no unread threads.
 - Fixed an issue where configuration changes could not be saved in the **System Console** in some cases.
 - Fixed typos in some translations that caused some in-product links to be broken.

#### API Changes
 - Added new API endpoints:
 
	  - ``GET /api/v4/users/me/top/dms``
	  - ``GET /api/v4/users/me/top/threads``
	  - ``GET /api/v4/teams/:team_id/top/team_members``
	  - ``GET /api/v4/teams/:team_id:/top/threads``
 - Added ``first_inaccessible_post_time`` to post API responses.
 - Updated permissions of the ``api/v4/posts/{post_id:[A-Za-z0-9]+}/thread`` endpoint. If compliance is enabled, a user can on longer view threads in a public channel they are not a member of.
 - Adds query parameter 'include_deleted' to endpoint: {{[http://your-mattermost-url.com/api/v4/posts/{post_id}/files/info}}](http://your-mattermost-url.com/api/v4/posts/%7Bpost_id%7D/files/info%7D%7D).

### Go Version
 - v7.3 is built with Go ``v1.18.1``.

### Open Source Components
 - Added ``@floating-ui/react-dom-interactions`` to https://github.com/mattermost/mattermost-webapp.

### Known Issues

>  - The new Insights feature has some performance costs that we are working to optimize.

Reviewer 2: Do we have any docs we can link to to explain how to disable the feature in case performance is degraded?

Author: We had recently added a note here https://docs.mattermost.com/install/self-managed-changelog.html#insights-beta-enterprise-and-professional, but I've also added that information to the known issues section now.

Reviewer 2: Ah ok, perfect. Thanks :)

 - Users can now request status updates (Professional).

#### Boards
 - All the boards you’re currently a member of from your current team will appear on the sidebar without needing to switch workspaces.
 - Organize boards on the sidebar with custom categories.
 - Press CTRL+K/CMD+K to find additional boards.
 - Navigate between teams in Boards with the new team switcher.
 - Set board and template permissions in the new **Share** setting.
 - Link boards to channels to automatically grant board permissions to channel members.
 - See [the documentation](https://docs.mattermost.com/welcome/whats-new-in-v72.html) for more details.

#### Calls
 - Added support for standalone Calls server and Kubernetes (Enterprise).
 
#### New Insights Widgets
 - Added four new [Insights widgets](https://docs.mattermost.com/welcome/insights.html): Most Active Direct Messages, Least Active Channels, Top Playbooks, and New Team Members.

### Improvements

#### User Interface (UI)
 - Added Calls keyboard shortcuts to the **Keyboard shortcuts** help modal.
 - Updated the "Contact Sales" link to ``mattermost.com/pl/contact-sales`` and update the pricing modal user interface.

>  - Introduced a new ``/marketplace command`` that brings up the marketplace modal for the Admin, and changed the ``/help`` command so that it now keeps the user internal to Mattermost.

Reviewer 3: ```suggestion
 - Introduced a new ``/marketplace`` slash commend that brings up the marketplace modal for the Admin, and changed the ``/help`` command so that it now keeps the user internal to Mattermost.
```

 - Team unreads are now calculated based on the channel membership and threads only. Team membership is no longer taken into account.
 - For introducing Boards and Playbooks to new users, an “explore other tools in platform” item was added to the end user onboarding checklist.
 - Added the **Save** option to the post menu.
 - Only the most recent message is now marked as unread when marking a thread as unread from the Threads list.
 - Insights filters now persist instead of being reset to default when switching to channels and returning back to the Insights view.
 - Code blocks now have better support for language filetype extensions and are a smaller bundle size.
 - A Desktop App prompt is now always shown on first visit to a Mattermost server from an email notification.

>  - Search dropdown options now allow focusing with tab.

Reviewer 3: ```suggestion
 - Search dropdown options now allow focusing by pressing the tab key.
```

 - Downgraded Bulgarian language support to Beta.

#### Administration

>  - Added a **View Plan** button in plan card at **System Console > License**.

Reviewer 3: ```suggestion
 - Added a **View Plan** button within the plan card via **System Console > License**.
```

 - Started tracking the join time of team members and added a new API endpoint to retrieve information about team members who have joined during a given time.
 - Introduced an optional ``shouldRender`` function parameter to ``registerchannelHeaderMenuAction`` plugin function. This allows menu items to conditionally render depending on the current state prior to rendering.
 - Plugins can now hide plugin settings based on the server's hosting environment.

>  - Customers who are on a 30-day free trial are now notified 3 days before the trial ends.

Reviewer 3: ```suggestion
 - Customers who are on a 30-day free trial are now notified three days before the trial ends.
```

### Bug Fixes
 - Fixed an issue where muted channels with an at-mention were displayed under the **Unreads** section of the channel switcher.
 - Fixed an issue where starting a trial failed if ``SiteURL`` was not set.
 - Fixed an issue where reading a thread on the mobile app caused a negative mention count to display on the web app.

>  - Fixed an issue where the user profile image persisted after user account deletion.

Reviewer 3: ```suggestion
 - Fixed an issue where the user's profile image persisted after user account deletion.
```

 - Fixed an issue where exports generated via mmctl without attachments still included the file properties in the post, so they couldn't be imported.
 - Fixed an issue that caused a crash when unread posts were fetched.
 - Fixed an issue where updating a profile image and creating new emojis used multipart uploads when using S3 storage.
 - Fixed an issue where the input legend on the custom group modal was cut off in Chrome.
 - Fixed an issue where the **Disable post formatting** setting was hidden when the advanced text editor was enabled.
 - Fixed an issue where we didn’t fall back to the user's default picture if a profile picture failed to load.
 - Fixed an issue where disabling a WebApp plugin from its configuration page resulted in the radio button reverting to ``true``.
 - Fixed an issue where the cursor sometimes jumped to the center channel textbox when the right-hand side was open.
 - Fixed an issue where closing the right-hand side also closed the edited post in the center channel.

>  - Fixed an issue where clicking "Try free now" opened the top 3 enterprise features instead of the "Your trial has started" modal.

Reviewer 3: ```suggestion
 - Fixed an issue where selecting "Try free now" opened the top three enterprise features instead of the "Your trial has started" modal.
```
